plugins/auth-node/src/IdentityClient.ts

```diff
@@ -51,8 +54,10 @@ export class IdentityClient {
   }) {
     this.discovery = options.discovery;
     this.issuer = options.issuer;
-    this.keyStore = new JWKS.KeyStore();
-    this.keyStoreUpdated = 0;
+    this.discovery.getBaseUrl('auth').then(url => {
```

Comment:
`getBaseUrl` needs to be called on every or close to every request, it can not be assumed that it returns a static URL. Iirc that's the reason we had the existing implementation rather than using the remote set

Reply:
The `createRemoteJWKSet` function was added to [`panva/jose`](https://github.com/panva/jose/commits/v3.0.0/docs/functions) in v3.0.0, so it might not have been considered in the initial implementation.

I updated `IdentityClient` to make a discovery call prior to authenticating and added a test that the updated URL is used.

Reply:
Removed the `async` call from the constructor